File: coding-patterns/00-warmup/001.py
# Contain Duplicate (Easy)

class Solution:
    def containsDuplicate(self, nums):
      count = {}
      for num in nums:
        if num not in count.keys():
          count[num] = 0
        count[num] += 1

      for num in nums:
        if count[num] > 1:
          return True
          
      return False

File: coding-patterns/00-warmup/001.cpp
// Contains duplicate (Easy)

#include <unordered_set>
#include <iostream>
#include <vector>
#include <unordered_map>

using namespace std;

class Solution
{
public:
  bool containsDuplicate(vector<int> &nums) {
    unordered_map<int, int> counter;
    for (auto num : nums) {
      counter[num];
      if (counter.count(num)) 
        counter[num]++;
      else counter[num] = 1;
    }

    for (auto p : counter) {
      if (p.second > 1)
        return true;
    }
    return false;
  }
};
